src/auto_code_show_messages.py

```diff
@@ -64,43 +64,37 @@ def auto_code_show_messages(cls, user, data, icr_output_dir, coda_output_dir):
             # This test works because the only codes which have been applied at this point are TRUE_MISSING.
             # If any other coding is done above, this test will need to change
             for td in data:
-                if plan.coded_field not in td:
+                if plan.raw_field in td:
+                    print(plan.raw_field)
                     rqa_messages.append(td)
-                else:
-                    assert len(td[plan.coded_field]) == 1
-                    assert td[plan.coded_field][0]["CodeID"] == \
-                        plan.code_scheme.get_code_with_control_code(Codes.TRUE_MISSING).code_id
-
-                icr_messages = ICRTools.generate_sample_for_icr(
-                    rqa_messages, cls.ICR_MESSAGES_COUNT, random.Random(cls.ICR_SEED))
                 
-                icr_output_path = path.join(icr_output_dir, plan.icr_filename)
-                with open(icr_output_path, "w") as f:
-                    TracedDataCSVIO.export_traced_data_iterable_to_csv(
-                        icr_messages, f, headers=[plan.run_id_field, plan.raw_field]
-                    )
-        
+            icr_messages = ICRTools.generate_sample_for_icr(
+                rqa_messages, cls.ICR_MESSAGES_COUNT, random.Random(cls.ICR_SEED))
+                
+            icr_output_path = path.join(icr_output_dir, plan.icr_filename)
+            with open(icr_output_path, "w") as f:
+                TracedDataCSVIO.export_traced_data_iterable_to_csv(
+                    icr_messages, f, headers=[plan.run_id_field, plan.raw_field]
+                )
+    
         # Output  Follow up Survey messages for ICR
         IOUtils.ensure_dirs_exist(icr_output_dir)
         for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
```

Comment:
Think this could be merged with the previous loop? If not, please explain why in a comment

Reply:
similar loops merged thanks!